src/Routers/UsersRouter.js

```diff
@@ -130,7 +133,48 @@ export class UsersRouter extends ClassesRouter {
               delete user.authData;
             }
           }
-
+          const mfaenabled = req.config.multiFactorAuth || {};
+          if (mfaenabled.enabled && recoveryTokens && user._mfa) {
+            const mfaRecTokens = user._mfa_recovery;
+            let firstAllowed = false;
+            let secondAllowed = false;
+            for (const recToken of mfaRecTokens) {
+              const setAllowedFromMatch = async (recoveryToken, first) => {
+                const doesMatch = await passwordCrypto.compare(recoveryToken, recToken);
+                if (!doesMatch) {
+                  return;
+                }
+                if (first) {
+                  firstAllowed = true;
+                } else {
+                  secondAllowed = true;
+                }
+              };
+              await setAllowedFromMatch(recoveryTokens.substring(0, 20), true);
+              await setAllowedFromMatch(recoveryTokens.substring(21, 41));
+            }
+            if (!firstAllowed || !secondAllowed) {
+              throw new Parse.Error(212, 'Invalid MFA recovery tokens');
+            }
+            await req.config.database.update(
+              '_User',
+              { username: user.username },
+              { _mfa: null, MFAEnabled: false, _mfa_recovery: null }
+            );
+            user.MFAEnabled = false;
+          } else if (mfaenabled.enabled && user._mfa) {
+            if (!token) {
+              throw new Parse.Error(211, 'Please provide your MFA token.');
```

Comment:
Requires Parse JS SDK reference for error code; check for possible inconsistency in error message for same error code.